compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -107,33 +108,54 @@ class Definitions {
    *      trait ImplicitFunctionN[T0,...,T{N-1}, R] extends Object with FunctionN[T0,...,T{N-1}, R] {
    *        def apply(implicit $x0: T0, ..., $x{N_1}: T{N-1}): R
    *      }
+   *
+   *  PhantomFunctionM traits follow this template:
+   *
+   *      trait PhantomFunctionM[T0,...T{N-1}, R] extends Object {
+   *        def apply($x0: T0, ..., $x{N_1}: T{N-1}): R
+   *      }
+   *
+   *      where M is a string of length N where each Char corresponds to a Ti
+   *      and Mi == '1' if the the parameter Ti is a phantom (Mi == '0' otherwise).
+   *
+   *  ImplicitPhantomFunctionM traits follow this template:
+   *
+   *      trait ImplicitPhantomFunctionM[T0,...,T{N-1}, R] extends Object with PhantomFunctionM[T0,...,T{N-1}, R] {
+   *        def apply(implicit $x0: T0, ..., $x{N_1}: T{N-1}): R
+   *      }
+   *
+   *      where M is a string of length N where each Char corresponds to a Ti
+   *      and Mi == '1' if the the parameter Ti is a phantom (Mi == '0' otherwise).
```

Comment:
the the

Do you think you could rephrase this not to leak the naming scheme? I also think `phantomicityOnChar` & `phantomicityOffChar` could be made private.